src/backend/distributed/planner/recursive_planning.c

```diff
@@ -337,10 +356,87 @@ RecursivelyPlanSubqueriesAndCTEs(Query *query, RecursivePlanningContext *context
 		RecursivelyPlanNonColocatedSubqueries(query, context);
 	}
 
+
+	PlannerRestrictionContext *plannerRestrictionContext =
+		context->plannerRestrictionContext;
+
+	List *rangeTableList = NIL;
+	GetRangeTableEntriesFromJoinTree((Node *) query->jointree, query->rtable,
+									 &rangeTableList);
```

Comment:
why do we need this instead of using rtable?

Reply:
I guess because we are only interested in range table entries that are on jointree, as in if there is a range table entry that is not in jointree but in rtable, we shouldn't even consider that.

Reply:
What kind of entries are those? The rtable basically describes the entries in the FROM. There might be some additional entries like views, but those are easily skipped.

Reply:
umm, the thought was that in case of WHERE FALSE etc jointree could be empty but it seems that it is not the case, so I am removing this.